cromshell

```diff
@@ -402,12 +405,178 @@ function populateWorkflowIdAndServerUrl()
+  local tmpTurtFile=$( makeTemp )
+  local turtFile=$( makeTemp )
+  local turtRFile=$( makeTemp )
+  turtle &> ${tmpTurtFile}
+  sed 's#^  ##g' ${tmpTurtFile} > ${turtFile}
+  rev ${turtFile} | sed -e 's$/$#$g' -e 's#\\#/#g' -e 's$#$\\$g' -e 's#(#)#g' -e 's#)#(#g' > ${turtRFile}
+
+  # Set up some loop variables:
+  local i=3
+  local incrementString="+1"
+  local curTurtFile=${turtFile}
+
+  # Let's loop while the ${flagFile} exists.
+  # Note that this will be spun off into its own thread:
+  while [ -f ${flagFile} ] ; do
+    # Create padding for turtle:
+    padding=$( printf "%${i}s" "" )
+
+    # Display our turtle, moved a little to the right:
+    cat ${curTurtFile} | sed -e "s#^#${padding}#g"
+    sleep 0.1
+
+    # Move the cursor back up to the top line of the turtle
+    # so we create the illusion of movement:
+    tput cuu 6
+
+    # Now we update our loop variables in a clever way
+    # so that the turtle will go back and forth.
+    let i=i${incrementString}
+    if [ $i -eq 15 ] ; then 
+      incrementString="-1" 
+      curTurtFile=${turtRFile}
+    elif [ $i -eq 0 ] ; then
+      incrementString="+1" 
+      curTurtFile=${turtFile}
+    fi
+
+  done & # The & here spins this off as a background thread/process
+}
+
+function waitOnSubmittedStatus() 
+{
+  local id=$1
+  local cromwellServerUrl=$2
+
+  echo "Waiting on status to progress from 'Submitted' ..."
+  echo
+
+  # Use tput to save the cursor position:
+  tput sc
+
+  # The turtle starts 9 rows above where the cursor is at this point,
+  # so if we need to move it, we have to move the cursor there before
+  # we output anything:
+  tput cuu 9
+
+  # Reserve the name for a file whose non-existence will cause 
+  # the following loop to exit.
+  # This is good because all temp files made with `makeTemp` 
+  # are automatically removed at exit, so this sub-process
+  # will always die gracefully.
+  local flagFile=$(makeTemp)
+  echo 'TEST' > ${flagFile}
+
+  # Kick off a turtle spinner...
+  _turtleSpinner ${flagFile}
+
+  # Now we check the status periodically and if we get a new one
+  # Other than `Submitted` we can tell the other process to finish.
+  local gotNewStatus=false
+  local isDone=false
+  local nChecks=0
+  local statusFile=$( makeTemp )
+  local r
+  while ! $isDone ; do
+    # Wait for a little bit so we don't kill the server:
+    sleep 2
+
+    # Check the status of our last submission:
+    status ${id} ${cromwellServerUrl} &> ${statusFile}
+    r=$?  
+    grep -q '^[ \t]*"status": "Submitted",' ${statusFile}
+
+    # We could no longer field the status as `Submitted`, so we're done.
+    if [ $? -eq 1 ] && [ $r -ne 2 ] ; then
+      gotNewStatus=true
+    fi
+
+    # Check if our status changed yet:   
+    if $gotNewStatus ; then
+      isDone=true
+ 
+      # Create the flag file to end the display process:
+      rm -f ${flagFile}
+
+      # Wait for the _turtleSpinner to exit:
+      wait
```

Comment:
`wait` without an argument will wait for all sub-processes to finish before unblocking.

Documentation for osx:
```
wait: wait [n]
    Wait for the specified process and report its termination status.  If
    N is not given, all currently active child processes are waited for,
    and the return code is zero.  N may be a process ID or a job
    specification; if a job spec is given, all processes in the job's
    pipeline are waited for.
```

Documentation for Linux:
```
wait: wait [id]
    Wait for job completion and return exit status.
    
    Waits for the process identified by ID, which may be a process ID or a
    job specification, and reports its termination status.  If ID is not
    given, waits for all currently active child processes, and the return
    status is zero.  If ID is a a job specification, waits for all processes
    in the job's pipeline.
    
    Exit Status:
    Returns the status of ID; fails if ID is invalid or an invalid option is
    given.
```

There is no race condition - the only subprocess at this point is the `_turtleSpinner`.  We wait here while the `_turtleSpinner` loops back and learns that the `flagFile` is gone (because we just removed it).